frontend/src/container/PipelinePage/ListOfPipelines.tsx

```diff
@@ -0,0 +1,408 @@
+		setChildDataSource(processorData);
+	};
+
+	const getExpandIcon = (
+		expanded: boolean,
+		onExpand: (
+			record: PipelineColumnType,
+			e: React.MouseEvent<HTMLElement>,
+		) => void,
+		record: PipelineColumnType,
+	): JSX.Element => {
+		if (expanded) {
+			return <DownOutlined onClick={(e): void => onExpand(record, e)} />;
+		}
+		return <RightOutlined onClick={(e): void => onExpand(record, e)} />;
+	};
+
+	const getFooterElement = (): JSX.Element => (
+		<Button
+			type="link"
+			onClick={(): void => setActionType('add-pipeline')}
+			style={modalFooterStyle}
+			icon={<PlusOutlined />}
+		>
+			{t('add_new_pipeline')}
+		</Button>
+	);
+
+	const handleModalCancelAction = (): void => {
+		setActionType(undefined);
+		formRef?.current?.resetFields();
+	};
+
+	return (
+		<div>
+			{contextHolder}
+			<Pipline
+				isActionType={isActionType}
+				setActionType={setActionType}
+				selectedRecord={selectedRecord}
+				setDataSource={setDataSource}
+				formRef={formRef}
+				handleModalCancelAction={handleModalCancelAction}
+			/>
+			<Processor
+				isActionType={isActionType}
+				setActionType={setActionType}
+				selectedRecord={selectedRecord}
+				setChildDataSource={setChildDataSource as () => Array<SubPiplineColumsType>}
+				formRef={formRef}
+				handleModalCancelAction={handleModalCancelAction}
+			/>
+			<Container>
+				<DndProvider backend={HTML5Backend}>
+					<Table
+						columns={pipelineColumns}
+						expandedRowRender={expandedRow}
+						expandable={{
+							expandedRowKeys: activeExpRow,
+							expandIcon: ({ expanded, onExpand, record }): JSX.Element =>
+								getExpandIcon(expanded, onExpand, record),
+							onExpand: (expanded, record): void => getDataOnExpand(expanded, record),
+						}}
+						components={tableComponents}
+						dataSource={dataSource.map((item) => ({
+							...item,
+							key: item.orderid,
+						}))}
+						// eslint-disable-next-line @typescript-eslint/no-explicit-any
+						onRow={(_record, index): React.HTMLAttributes<any> => {
+							const attr = {
+								index,
+								moveRow: movePipelineRow,
+							};
+							// eslint-disable-next-line @typescript-eslint/no-explicit-any
+							return attr as React.HTMLAttributes<any>;
+						}}
+						footer={(): React.ReactElement => getFooterElement()}
+					/>
+				</DndProvider>
+			</Container>
+		</div>
+	);
+}
+
+interface ListOfPipelinesProps {
+	isActionType: string | undefined;
+	setActionType: (b: string | undefined) => void;
+}
+
+export interface PipelineOperatorsType {
+	type: string;
+	name: string;
+	id: string;
+	field?: string;
+	parse_from?: string;
+	parse_to?: string;
+	output?: string;
+	pattern?: string;
+	trace_id?: { parse_from: string };
+	span_id?: { parse_from: string };
+	trace_flags?: { parse_from: string };
+}
+
+export interface PipelineColumnType {
+	orderid: number;
+	uuid: string;
+	createdAt: string;
+	createdBy: {
+		username: string;
+		email: string;
+	};
+	updatedAt: string;
+	updatedBy: {
```

Comment:
updatedBy and createdBy are using same object can we can use the same named interface